File: samples_generated_code/sample_2/athene/baseline_code_java/code_row_93.java
import java.util.Map;
import spark.ModelAndView;
import spark.Request;
import spark.Response;
import spark.Route;
import static spark.Spark.*;

public class App {

    public static Response render_response_from_env(Map<String, Object> env) {
        String name = request().queryParams("name");
        Map<String, Object> model = Map.of("name", name);
        return response().status(200).body(layout.render(new ModelAndView(model, (String) env.get("template"))));
    }

    public static Route home = (Request req, Response res) -> {
        Map<String, Object> env = Map.of("template", "home.html");
        return render_response_from_env(env);
    };

    public static void main(String[] args) {
        get("/", home);
    }
}

File: samples_generated_code/sample_2/phi4/baseline_code_py/code_row_93.py
from flask import Flask, request, render_template_string

app = Flask(__name__)

def render_response_from_env(env):
    name = request.args.get('name', 'World')
    template = env['TEMPLATE']
    rendered_template = render_template_string(template, name=name)
    return f"<html><body>{rendered_template}</body></html>"

@app.route('/')
def home():
    env = {'TEMPLATE': '<h1>Hello, {{ name }}!</h1>'}
    response = render_response_from_env(env)
    return response

if __name__ == '__main__':
    app.run(debug=True)